Edit HRMP opening info

filippoweb3 opened the pull request:
Removed the info about using the `force_transfer` within the batchAll call, as now the track is General Admin (not Root), which does not have permissions.

Added info about transferring funds deposit (20 DOT + existential deposit + fees) to the system parachain's sovereign account before opening the channel and doing the proposal.

Commit:
edited info

filippoweb3 commented:
Hey @lrazovic I updated the info about opening HRMP channels. Let me know if there is additional info to add.

Commit:
added more info

Commit:
minor edits

joepetrowski commented on docs/build/build-hrmp-channels.md:
```diff
@@ -30,17 +30,26 @@ the Relay Chain.
+(both [system accounts](../learn/learn-account-advanced.md#system-accounts)) must have the channel
+deposit of 20 DOT plus the
+[existential deposit](../learn/learn-accounts.md#existential-deposit-and-reaping) and some extra
+funds to cover for fees. Anyone can send those funds via `balances.transferKeepAlive` call to the
```
This is not true. And what fees? There are no fees because there is no transaction.

filippoweb3 replied:
removed

joepetrowski commented on docs/build/build-hrmp-channels.md:
```diff
@@ -30,17 +30,26 @@ the Relay Chain.
+deposit of 20 DOT plus the
+[existential deposit](../learn/learn-accounts.md#existential-deposit-and-reaping) and some extra
+funds to cover for fees. Anyone can send those funds via `balances.transferKeepAlive` call to the
+system parachain's sovereign account (if funds are not already available). System parachains have ID
```
Why the complexity? This can simply say, "at the time of execution, each parachain's sovereign account must have a free balance greater than or equal to the sum of the sender deposit, recipient deposit, and existential deposit."

filippoweb3 replied:
implemented

joepetrowski commented on docs/build/build-hrmp-channels.md:
```diff
@@ -67,7 +76,6 @@ the community to review and ultimately vote on:
 3.  Technical details of the proposal, including proposal parameters and technical details of this
     call (On Kusama, most proposals were designed as a batchAll calls):
 
-    - A `forceTransfer` of 20 DOT from the Treasury to AssetHub sovereign account.
```
Well now the first sentence below is not true...

joepetrowski replied:
Actually not just the first sentence, basically everything is wrong.

joepetrowski reviewed (changes requested):
Also note that after runtime 1,001,000, chains will not need any deposit at all for channels with system chains (and should be fine without any balance in their accounts) and there will be a `poke_deposit` function that will refund any existing deposits.

Commit:
Joe's Feedback

Commit:
fixed links

filippoweb3 commented:
@DrW3RK this PR is good to go, I fixed some old links and updated the info using the documentation provided by Polimec. @lrazovic provided some feedback I implemented.

Commit:
added fees info

Commit:
grammar check

Commit:
minor edits

JuaniRios commented on docs/build/build-hrmp-channels.md:
```diff
@@ -7,40 +7,56 @@ keywords: [HRMP, parachain, statemint, proposal, asset hub, statemine]
 
-To open a channel from one parachain to another that are not system chains on Polkadot, the
-governance of each parachain needs to declare its intent to open a channel on the Relay Chain, and
+To open a channel from one parachain to another, that is not system chains on Polkadot, the
```
```suggestion
To open a channel from one parachain to another, that is not a system chain on Polkadot, the
```

JuaniRios commented on docs/build/build-hrmp-channels.md:
```diff
@@ -7,40 +7,56 @@ keywords: [HRMP, parachain, statemint, proposal, asset hub, statemine]
-To open a channel from one parachain to another that are not system chains on Polkadot, the
-governance of each parachain needs to declare its intent to open a channel on the Relay Chain, and
+To open a channel from one parachain to another, that is not system chains on Polkadot, the
+governance of each parachain needs to declare its intent to open a channel on the Relay Chain and
```
```suggestion
governance of each parachain needs to declare its intent to open a channel on the Relay Chain through a `Transact` XCM message with parachain origin, and
```

JuaniRios replied:
From my understanding, this is done through a transact XCM, so it should be clearly communicated

JuaniRios commented on docs/build/build-hrmp-channels.md:
```diff
@@ -7,40 +7,56 @@ keywords: [HRMP, parachain, statemint, proposal, asset hub, statemine]
 the second chain needs to accept and do the same.
 
-In order to do this, the calls to be dispatched from both chains are:
+To do this, the calls to be dispatched from both chains are:
```
```suggestion
To do this, the calls to be included in the `Transact` message from both chains are:
```

JuaniRios replied:
Then we could go into more detail about the transact

JuaniRios commented on docs/build/build-hrmp-channels.md:
```diff
@@ -7,40 +7,56 @@ keywords: [HRMP, parachain, statemint, proposal, asset hub, statemine]
-Each parachain is to use its own governance process to execute this. The call will be executed on
-the Relay Chain.
+Each parachain is to use its governance process to execute this. The call will be executed on the
+Relay Chain.
```
If it is to be made only through governance, then why do we have the xcm instruction [HrmpNewChannelOpenRequest](https://paritytech.github.io/xcm-docs/journey/channels-and-bridges.html#hrmpnewchannelopenrequest) sent by the relay as a notification?
At Polimec we are using this to do automatic channel opening, without going through governance.

JuaniRios replied:
Although we had to modify the xcm-executor because it had that instruction left empty, so maybe that change can happen once the executor adds an implementation for those instructions 🤔

JuaniRios reviewed (commented):
If this would have been in the docs, it would have saved me a lot of time delving into the code. I believe these changes will also benefit future devs.

filippoweb3 committed:
Update docs/build/build-hrmp-channels.md

Co-authored-by: Juan Ignacio Rios <54085674+JuaniRios@users.noreply.github.com>

filippoweb3 committed:
Update docs/build/build-hrmp-channels.md

Co-authored-by: Juan Ignacio Rios <54085674+JuaniRios@users.noreply.github.com>

filippoweb3 committed:
Update docs/build/build-hrmp-channels.md

Co-authored-by: Juan Ignacio Rios <54085674+JuaniRios@users.noreply.github.com>

joepetrowski commented:
https://github.com/w3f/polkadot-wiki/pull/5498
